### PR TITLE
Fix cupy not working in thread other than one imported cupy

### LINKED ISSUE #1579: bug for 4.4.0 release. Can not working on multi threading.
This [merge](https://github.com/cupy/cupy/pull/1350/files) has bug when working on multi threading. 

image

Appear this issue after upgrade to 4.4.0. I had to rollback my package to 4.3.0 now.

### PR DESCRIPTION
**Pull request by kmaehashi**
This fixes #1579.

### CODE REVIEW

**Comment by kmaehashi** [comment]
@asi1024 (and possibly @okuta) Could you check this?

**Comment by kmaehashi** [comment]
Jenkins, test this please.

File: cupy/core/fusion.py [hunk 1]
```diff
@@ -753,15 +752,15 @@ def __repr__(self):
         return '<Fusion \'{}\'>'.format(self.name)
 
     def __call__(self, *args, **kwargs):
-        if _thread_local.history is None:
+        if hasattr(_thread_local, 'history'):
```
Comment (asi1024) [inline cupy/core/fusion.py]: This condition should be `not hasattr(_thread_local, 'history')`.

Comment (kmaehashi) [reply cupy/core/fusion.py]: Thanks, fixed.

**Comment by kmaehashi** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
LGTM.

**Pull request merged**